src/k8s-engine/lima.ts

```diff
@@ -241,6 +237,21 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     return this.#desiredPort;
   }
 
+  protected async ensureArchitectureMatch() {
+    if (os.platform().startsWith('darwin')) {
+      const fileArch = this.arch === 'aarch64' ? 'arm64' : this.arch;
+      const { stdout } = await childProcess.spawnFile(
+        'file', [this.limactl],
+        { stdio: ['inherit', 'pipe', console] });
+
+      if (!stdout.trim().match(`executable ${ fileArch }$`)) {
+        const otherArch = { aarch64: 'x86_64', x86_64: 'aarch64' }[this.arch];
+
+        throw new K8s.KubernetesError('Fatal Error', `Rancher Desktop for ${ otherArch } does not work on ${ this.arch }.`, true);
```

Comment:
It might be nice to say something like `Apple M1` instead; this depends on Apple's marketing…

Reply:
I wanted to use the same arch name we use in the filename, `Rancher Desktop.aarch64.dmg` and `Rancher Desktop.x86_64.dmg`, so users can map between the error message and the disk image file.

Reply:
Ah; given _that_ this should be okay. (Maybe add a comment about it if you feel like it.)